### ISSUE TITLE
assigning member to ModuleType

### ISSUE THREAD

**Issue by dumbPy** [microsoft/pyright#462]
**Describe the bug**
I am trying to set a module level variable from within a method. module accesed from `sys.modules` has type `ModuleType`. Settings variable to this raises an error
```
Cannot assign member 'bar' for type 'ModuleType' 
  Member 'bar' is unknown
```

**To Reproduce**
```
project_root
├── foo
│  └── __init__.py
└── run.py
```
``` python
# Inside foo/__init__.py
import sys
bar = "bar not set yet"
def set_bar():
    this_module = sys.modules[__name__] # <-- this_module of type ModuleType
    this_module.bar = "new bar value" # <-- pyright error raised in this line

# run.py
import foo
print(foo.bar)
foo.set_bar()
print(foo.bar)
```
**Expected behavior**
Should not raise an error

**VS Code extension or command-line**
extension v 1.1.16

**Comment by erictraut** [comment]
Pyright is doing the right thing in this case.

`sys.modules` is defined as `modules: Dict[str, ModuleType]`, so `this_module` is of type `ModuleType`.

`ModuleType` is a class defined in the stdlib type stubs, and it doesn't have a `__setattr__` method, so writing to unknown attributes is not permitted. So Pyright is correctly reporting the error.

The way you're trying to modify the module's namespace is very hacky. From within the module, you're looking up itself within the global system module table, then you're dynamically modifying its global namespace.

The straightforward and clean way to set a module-level variable from within that module is:
```
def set_bar():
    global bar
    bar = "new bar value"
```

**Comment by dumbPy** [comment]
Ah.. I was trying to avoid global variables due to "Global Variables are a bad idea", without realizing that I am actually settings and overwriting a global variable in a hacky way. Thanks Again!!
